python/cudf/dataframe/column.py

```diff
@@ -407,16 +407,42 @@ def __getitem__(self, arg):
         else:
             raise NotImplementedError(type(arg))
 
+    def masked_assign(self, value, mask):
+        """Assign a scalar value to a series using a boolean mask
+        df[df < 0] = 0
+
+        Parameters
+        ----------
+        value : scalar
+            scalar value for assignment
+        mask : cudf Series
+            Boolean Series
+
+        Returns
+        -------
+        cudf Series
+            cudf series with new value set to where mask is True
+        """
+
+        # need to invert to properly use gpu_fill_mask
+        mask._column._invert()
+        mask_gpu = mask.as_mask()
+        out = cudautils.fill_mask(data=self.data.to_gpu_array(),
+                                  mask=mask_gpu,
+                                  value=value)
+
+        return self.replace(data=Buffer(out), mask=None, null_count=0)
+
     def fillna(self, value):
         """Fill null values with ``value``.
 
         Returns a copy with null filled.
         """
         if not self.has_null_mask:
             return self
-        out = cudautils.fillna(data=self.data.to_gpu_array(),
-                               mask=self.mask.to_gpu_array(),
-                               value=value)
+        out = cudautils.fill_mask(data=self.data.to_gpu_array(),
```

Comment:
I think this should be changed back to fillna?

Reply:
I thought you said fillna was going to be a libcudf call in the future and I should build a new function?

Reply:
For your `masked_assign` function, but we shouldn't change the cudautils function called by the `column.fillna` function.

Reply:
Oh!  My apologies, that was a mistake.  Sorry sorry